src/components/BookPage/index.js

```diff
@@ -47,12 +47,17 @@ function BookPage() {
         console.log(info.data.reviews.length);
       })
       .catch((error) => console.error("Error fetching data", error));
-  }, [id]);
+  }
+  useEffect(() => {
+    fetchBookData()
+  }, [id])
   function handleReviewSubmit(newReview) {
-    setReviews([...reviews, newReview])}
+    setReviews([...reviews, newReview])
```

Comment:
You don't need to do this anymore. Because the form adds the new review, and then you re-fetch the book, you naturally get the new review.

No need to setReviews, or to pass the review into this function.

I would actually remove this function entirely, and rather than passing this down to the review form, instead pass down fetchBookData and call that upon submission